File: eecs168/2020spring/video-lectures/2020.04.22.classes-part-2/Triangle.cpp

//cpp files for classes are implementation files
//All methods are defined in the cpp file


//First thing to do in cpp is include the header file
#include "Triangle.h"

Triangle::Triangle()
{
	//Assigning base and height initial values
	base = 0;
	height = 0;
}

//This method belongs to the Triangle
double Triangle::area()
{
	//base and height's values will be 
	//determined based on which instance/object
	//is doing the calling
	double ans = 0;
	ans = 0.5*base*height;
	return(ans);
}

File: eecs168/2023summer/2023.07.17.classes-magic-methods/triangledemo/triangle.py
#triangle.py

class Triangle:
    def __init__(self, base=0, height=0):
        self.base = base
        self.height = height

    def area(self):
        answer = (1/2)*self.base*self.height
        return answer

    def __lt__(self, rhs):
        return self.area() < rhs.area()

    def __eq__(self, rhs):
        return self.base == rhs.base and self.height == rhs.height
    
